Expose MMFF aromaticity model for SetAromaticity

JLVarjo opened the pull request:
Invoking MMFF optimization sets atom and bond aromaticity flags according to MMFF aromaticity model, but this model is not accessible via `MolOps::SetAromaticity`, thus being a somewhat hidden one. My workaround so far has been to call `MMFFOptimizeMolecule` with zero allowed steps, but this obviously performs the whole MMFF initialization which is at least unnecessary work. This PR is a suggestion to expose the MMFF aromaticity model as one of the explicitly supported models of `MolOps::SetAromaticity`. I had to move `setMMFFAromaticity` under MolOps due to linker issues.

Another question is, if MMFF altering the Mol object aromaticity is bit unexpected and dangerous in general (despite being documented). It seems that any other properties of Atoms and Bonds (except coordinates of course) are not touched in MMFF GO. An option would be to store the MMFF aromaticity flags in MMFF internal atom and bond properties only, which would then leave Mol aromaticity unchanged during MMFF geometry optimization. But this is up to you to discuss, I can imagine reasoning for either option.

greglandrum commented:
@ptosco : I think the idea of having the MMFF aromaticity model available like the other models makes sense. What's your take?

ptosco commented:
@greglandrum Yes, I think it makes sense.
@JLVarjo Regarding the fact that MMFF modifies aromaticity flags, I agree it would be much better not to, and I would not do it that way today - when I originally did that implementation >10 years ago I did not fully realize the implications.

greglandrum commented on Code/GraphMol/ForceFieldHelpers/MMFF/AtomTyper.h:
```diff
@@ -217,7 +217,6 @@ RDKIT_FORCEFIELDHELPERS_EXPORT bool areAtomsInSameRingOfSize(
     const ROMol &mol, const unsigned int ringSize, const unsigned int numAtoms,
     ...);
 RDKIT_FORCEFIELDHELPERS_EXPORT unsigned int sanitizeMMFFMol(RWMol &mol);
-RDKIT_FORCEFIELDHELPERS_EXPORT void setMMFFAromaticity(RWMol &mol);
```
Rather than completely removing this, which may break others' code, please mark it as deprecated (and add a note in the `Deprecated code` section of README.md) and call the `MolOps` version from it.

JLVarjo replied:
Done! Please check if the deprecation markings are ok, tried my best to copy how they are done elsewhere

greglandrum reviewed (changes requested):
The core of the changes look fine.
Before this can be merged, we need a test to be added to `GraphMol/catch_molops.cpp` that verifies that the new aromaticity option actually works. Feel free to ask if you have questions about how to do this.

I also made a backwards-compatibility suggestion.

JLVarjo commented:
Added one simple test to `GraphMol/molopstest.cpp`, from where I found other SetAromaticity-related tests. Please let me know what kind of test you want me to add to `GraphMol/catch_molops.cpp`, and if any other tests are required.

Also noticed that I forgot to add a python wrapper for the new AromaticityModel, this is also added in the commit.

greglandrum reviewed (approved):
LGTM

greglandrum commented:
> Added one simple test to `GraphMol/molopstest.cpp`, from where I found other SetAromaticity-related tests. Please let me know what kind of test you want me to add to `GraphMol/catch_molops.cpp`, and if any other tests are required.

Adding tests to the older file is also fine; I just find it easier to add catch tests. 


> Also noticed that I forgot to add a python wrapper for the new AromaticityModel, this is also added in the commit.

good catch!

greglandrum commented:
Thanks @JLVarjo

JLVarjo commented:
Thanks for the pull!

One more question regarding possible follow-up. Would you be interested of a change that would make the Mol object aromaticity flags to remain untouched during GO, or would this break too much backwards compatibility? If yes, I can try to make such change at some point.
